update.php

```diff
@@ -500,6 +500,43 @@
     }
 }
 
+// 20150401 Add mail method to database (SMTP/sendmail)
+$sql = "SELECT COUNT(*) AS confcnt FROM config";
+$req = $pdo->prepare($sql);
+$req->execute();
+$confcnt = $req->fetch(PDO::FETCH_ASSOC);
+
+if ($confcnt['confcnt'] < 20) {
+    $mail_method = 'sendmail';
+    // check if an smtp server was set
+    $sql = "SELECT * FROM config";
+    $req = $pdo->prepare($sql);
+    $req->execute();
+    $config_items = [];
+    while ($show = $req->fetch()) {
+        array_push($config_items, $show);
+    }
+
+    if ($config_items['smtp_address'] !== '') {
+        $mail_method = 'smtp';
+        $smtp_username = filter_var($config_items['smtp_username'], FILTER_VALIDATE_EMAIL);
```

Comment:
You are absolutely right, but that's how it's implemented right now in next-branch ;-) Sending emails via SMTP in current next-branch fails because "smtp_username" is used  as sender address (see e.g. admin_exec.php, line 250). This does _not_ work, because Swift mailer expects an email address there. By that, it's implicated, that smtp_username has to be a valid email address. That's why I introduced "mail_from" (https://github.com/Athemis/elabftw/commit/8814422fe4269cfc46683521d96b00f43bccc976) which is used instead of smtp_username as sender address.